include/rdma/fi_var.h

```diff
@@ -40,6 +40,16 @@
 extern "C" {
 #endif
 
+struct fi_var_t {
+	const struct fi_provider *provider;
+	char *var_name;
+	char *help_string;
+
+	char *env_var_name;
+
+	struct fi_var_t *next;
+};
```

Comment:
Unless these structures are allocated per call to fi_getvars(), this structure is either racy or limits what can be done internally.  There's no way for libfabric to modify this list safely if an app can be walking through these structures because of a lack of synchronization.

We need a different approach for iterating over this data.  Regardless of how we intend this data to be used, it is exposed through the public API.  An app can (and eventually will) walk this list and look at these structures.